### PR TITLE
Consolidate import and usage of itertools

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

**Comment by uranusjr** [comment]
If we want to consolidate (not strictly necessary) I’d prefer the full `itertools` name. `it` is used in some places as a variable (short for `iterator`).

File: airflow/www/views.py [hunk 1]
```diff
@@ -3229,8 +3229,7 @@ def duration(self, dag_id: str, session: Session = NEW_SESSION):
 
         # we must group any mapped TIs by dag_id, task_id, run_id
         mapped_tis = set()
-        tis_grouped = itertools.groupby(task_instances, lambda x: (x.dag_id, x.task_id, x.run_id))
-        for _, group in tis_grouped:
+        for _, group in it.groupby(task_instances, lambda x: (x.dag_id, x.task_id, x.run_id)):
```
Comment (kaxil) [inline airflow/www/views.py]: hmm ? why this change?

File: airflow/lineage/__init__.py [hunk 2]
```diff
@@ -142,7 +142,7 @@ def wrapper(self, context, *args, **kwargs):
                 _inlets = self.xcom_pull(
                     context, task_ids=task_ids, dag_id=self.dag_id, key=PIPELINE_OUTLETS, session=session
                 )
-                self.inlets.extend(i for i in itertools.chain.from_iterable(_inlets))
+                self.inlets.extend(itertools.chain.from_iterable(_inlets))
```
Comment (uranusjr) [inline airflow/lineage/__init__.py]: This can just be

```suggestion
                self.inlets.extend(i for it in _inlets for i in it)
```

but itertools has readability advantages.

File: scripts/ci/pre_commit/pre_commit_check_deferrable_default.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def iter_check_deferrable_default_errors(module_filename: str) -> Iterator[str]:
         args = node.args
         arguments = reversed([*args.args, *args.kwonlyargs])
         defaults = reversed([*args.defaults, *args.kw_defaults])
-        for argument, default in itertools.zip_longest(arguments, defaults, fillvalue=None):
+        for argument, default in zip(arguments, defaults):
```
Comment (uranusjr) [inline scripts/ci/pre_commit/pre_commit_check_deferrable_default.py:79]: These are not equivalent.

Comment (eumiro) [reply scripts/ci/pre_commit/pre_commit_check_deferrable_default.py:79]: In the subsequent line:

```python
            if argument is None or default is None:
                continue
```

all the lines that come from `zip_longest` but are ignored in `zip` are ignored anyway.

**Comment by potiuk** [comment]
Conflicts :(

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
